azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt

```diff
@@ -6,15 +6,31 @@ package com.azure.android.communication.ui.chat.redux.middleware.sdk
 import com.azure.android.communication.ui.chat.error.ChatStateError
 import com.azure.android.communication.ui.chat.error.ErrorCode
 import com.azure.android.communication.ui.chat.redux.Dispatch
+import com.azure.android.communication.ui.chat.redux.Store
 import com.azure.android.communication.ui.chat.redux.action.Action
 import com.azure.android.communication.ui.chat.redux.action.ChatAction
 import com.azure.android.communication.ui.chat.redux.action.ErrorAction
+import com.azure.android.communication.ui.chat.redux.action.NetworkAction
+import com.azure.android.communication.ui.chat.redux.state.NetworkStatus
+import com.azure.android.communication.ui.chat.redux.state.ReduxState
 import com.azure.android.communication.ui.chat.service.ChatService
 
 // Converts Redux Actions into SDK Calls
 // Redux -> Service
 internal class ChatActionHandler(private val chatService: ChatService) {
 
+    fun onAction(store: Store<ReduxState>, action: Action) {
```

Comment:
add ReduxState to the other onAction so we don't have 2 onActions in the ChatActionHandler

However, feel this action should be handled by a separate middleware that brings in 
NetworkAction.Connected and Emits a ChatAction that results in FetchMessages call.